### PR TITLE
mod : Crusthrough rework

### PR DESCRIPTION
**Pull request by namidaka**
Table Provided here : https://docs.google.com/spreadsheets/d/1A5ynImHS9oKFTJzyX9l2YUg8bHNyuh8T6uhfXxOBENA/edit?usp=sharing

### CODE REVIEW

File: src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs [hunk 1]
```diff
@@ -202,6 +202,45 @@ public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart,
         defenderStunMultiplier = 1f;
     }
 
+    // Consider reworking once https://forums.taleworlds.com/index.php?threads/missioncombatmechanicshelper-getdefendcollisionresults-bypass-strikemagnitudecalculationmodel.459379 is fixed
```
Comment (MrMendo) [inline src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs:205]: maybe declare this as a summary
could be easier to find it again.
Alternatively add a "TODO: " as prefix to search for todos in the future

File: src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs [hunk 2]
```diff
@@ -202,6 +202,45 @@ public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart,
...
+
+        float attackerPowerStrikeSkill = 3f * GetSkillValue(attackerAgent.Origin, CrpgSkills.PowerStrike);
+
+        float defenderStrengthSkill = GetSkillValue(defenderAgent.Origin, CrpgSkills.Strength);
+        float defenderShieldSkill = GetSkillValue(defenderAgent.Origin, CrpgSkills.Shield);
+        float defenderDefendPower = defendItem != null && defendItem.IsShield
+            ? Math.Max(defenderShieldSkill * 6 + 3, defenderStrengthSkill)
+            : defenderStrengthSkill;
+        int randomNumber = MBRandom.RandomInt(0, 1000);
+        return randomNumber / 10f < Math.Pow(attackerPowerStrikeSkill / defenderDefendPower / 2.5f, 1.8f) * 100f;
```
Comment (MrMendo) [inline src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs:241]: Do i not understand that or why do you calculate randomNumber / 10f?
is it not the same as having a randomInt of 0 - 100?

File: src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs [hunk 3]
```diff
@@ -202,6 +202,45 @@ public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart,
...
+        if (weaponComponentData == null
+            || isPassiveUsage
+            || !weaponComponentData.WeaponFlags.HasAnyFlag(WeaponFlags.CanCrushThrough)
+            || strikeType != StrikeType.Swing
+            || attackDirection != Agent.UsageDirection.AttackUp)
+        {
+            return false;
+        }
+
+        float attackerPowerStrikeSkill = 3f * GetSkillValue(attackerAgent.Origin, CrpgSkills.PowerStrike);
```
Comment (MrMendo) [inline src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs:233]: Rename variable to attackerPowerStrikeSkillPower
was a bit confused in Line 241 that we only take the raw attackerPowerStrikeSkill against the defenderDefendPower.

**Review by MrMendo — COMMENTED** [review]
mostly nitpick and a question

File: src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs (hunk 2 above)
Comment (namidaka) [reply src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs:241]: Not the same as the right part can be 82.5%
This gives 1 decimal point of precision

**Pull request merged**